onnxruntime/core/providers/rocm/math/matmul.cc

```diff
@@ -0,0 +1,232 @@
+}
+
+template <typename T>
+Status MatMul<T>::ComputeInternal(OpKernelContext* ctx) const {
+  typedef typename ToHipType<T>::MappedType HipT;
+
+  const Tensor* left_X = ctx->Input<Tensor>(0);
+  const Tensor* right_X = ctx->Input<Tensor>(1);
+
+  // Ignore the transpose flag if rank of input being 1.
+  // Be noted: numpy.transpose on vector does not change anything.
+  bool transa = trans_A_;
+  bool transb = trans_B_;
+  if (left_X->Shape().NumDimensions() == 1) {
+    transa = false;
+  }
+  if (right_X->Shape().NumDimensions() == 1) {
+    transb = false;
+  }
+
+  MatMulComputeHelper helper;
+  ORT_RETURN_IF_ERROR(helper.Compute(left_X->Shape(), right_X->Shape(), transa, transb));
+
+  Tensor* Y = ctx->Output(0, helper.OutputShape());
+
+  // Bail out early if the output is going to be empty
+  if (Y->Shape().Size() == 0)
+    return Status::OK();
+
+  const HipT alpha = ToHipType<T>::FromFloat(alpha_);
+  const HipT zero = ToHipType<T>::FromFloat(0.0f);
+
+  // hipblasOperation_t transA = transa ? HIPBLAS_OP_T : HIPBLAS_OP_N;
+  // hipblasOperation_t transB = transb ? HIPBLAS_OP_T : HIPBLAS_OP_N;
+  rocblas_operation transA = transa ? rocblas_operation_transpose : rocblas_operation_none;
+  rocblas_operation transB = transb ? rocblas_operation_transpose : rocblas_operation_none;
+  const int lda = transa ? static_cast<int>(helper.M()) : static_cast<int>(helper.K());
+  const int ldb = transb ? static_cast<int>(helper.K()) : static_cast<int>(helper.N());
+  const int ldc = static_cast<int>(helper.N());
+  int64_t stride_A, stride_B, stride_C, batch_count;
+
+  if (helper.OutputOffsets().size() == 1) {
+    // HIPBLAS_RETURN_IF_ERROR(hipblasGemmHelper(
+    //     Base::HipblasHandle(),
+    //     transB,
+    //     transA,
+    //     static_cast<int>(helper.N()),
+    //     static_cast<int>(helper.M()),
+    //     static_cast<int>(helper.K()),
+    //     &one,
+    //     reinterpret_cast<const HipT*>(right_X->template Data<T>()),
+    //     ldb,
+    //     reinterpret_cast<const HipT*>(left_X->template Data<T>()),
+    //     lda,
+    //     &zero,
+    //     reinterpret_cast<HipT*>(Y->template MutableData<T>()),
+    //     ldc));
+    ROCBLAS_RETURN_IF_ERROR(rocblasGemmHelper(
+        Base::RocblasHandle(),
+        transB,
+        transA,
+        static_cast<int>(helper.N()),
+        static_cast<int>(helper.M()),
+        static_cast<int>(helper.K()),
+        &alpha,
+        reinterpret_cast<const HipT*>(right_X->template Data<T>()),
+        ldb,
+        reinterpret_cast<const HipT*>(left_X->template Data<T>()),
+        lda,
+        &zero,
+        reinterpret_cast<HipT*>(Y->template MutableData<T>()),
+        ldc));
+    return Status::OK();
+  } else if (CanUseStridedBatchedGemm(left_X->Shape(), right_X->Shape(),
+                                      transa, transb, stride_A, stride_B, stride_C, batch_count)) {
+    // HIPBLAS_RETURN_IF_ERROR(hipblasGemmStridedBatchedHelper(Base::HipblasHandle(),
```

Comment:
Delete it?

Reply:
Plan to have a separate PR to replace all hipBLAS call with rocBLAS call.